File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [-int(1e9) for i in range(n)]
        result = -int(1e9)

        for i in range(n):
            if i == 0:
                dp[i] = nums[i]
                result = dp[i]
            else:
                dp[i] = max(nums[i], dp[i - 1] + nums[i])
                result = max(dp[i], result)

        return result

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int result = nums[0];
        int now = nums[0];

        for (int i = 1; i < nums.length; i++) {
            now = Math.max(nums[i], now + nums[i]);
            result = Math.max(result, now);
        }

        return result;
    }
}